.gitlab-ci.yml

```diff
@@ -1,27 +1,10 @@
-image: govcmsdev/gitlab-ci
-services:
-- docker:dind
-stages:
-- build
-- deploy
-build:
-  stage: build
-  script:
-    - "export DOCKER_HOST=tcp://localhost:2375"
-    - "docker network prune -f && docker network create amazeeio-network"
-    - "docker login -u gitlab-ci-token -p $CI_JOB_TOKEN gitlab-registry-production.govcms.amazee.io"
-    - "docker-compose up -d"
-    - "docker-compose exec -T test dockerize -wait tcp://mariadb:3306 -timeout 1m"
-    - |
-      docker-compose exec -T cli composer install
-      if ! grep --quiet '^MARIADB_DATA_IMAGE' .env; then
-        echo "MARIADB_DATA_IMAGE is not set. @see https://govcms.gov.au/wiki_vars#MARIADB_DATA_IMAGE"
-        docker-compose exec -T cli drush si -y govcms
-      fi
-      docker-compose exec -T cli govcms-deploy
-    - ahoy lint || true
-deploy:
-  stage: deploy
-  script:
-    - "echo \"curl -X POST -d projectName=$CI_PROJECT_NAME -d branchName=$CI_COMMIT_REF_NAME http://rest2tasks.lagoon.svc:3000/deploy\""
-    - "curl -X POST -d projectName=$CI_PROJECT_NAME -d branchName=$CI_COMMIT_REF_NAME http://rest2tasks.lagoon.svc:3000/deploy"
+# This file is locked for GovCMS SaaS.
+include:
+
+  # Starting inputs for the GitLab CI configuration.
```

Comment:
this comment has no usable meaning. It does not describe what these inputs are and what to do with them. Something like this would be more helpful: 

```
GitLab CI configuration per-project overrides to alter job properties.
```

Reply:
One thing to note is they are not overrides - you can't override the gitlab CI from this file the way the .gitlab-ci-main.yml is structure.

 We really want people to read the docs on this file, developers should have a technical interpretation of what an "input" is to a program. Users (if the file is unlocked) should not be under the assumption that they can overload this file with all sorts of stuff. It is a carefully curated list of arguments to a controlled CI framework.

The wiki for (less or non-technical) users will be "I want to disable linting." or "I want to enable my behat tests but not block deployments". And the response will be: go to this very simple file and change this very simple thing.

Reply:
@simesy the fact that you had to explain this to me here proves my point - it is not clear straight away what this is. The policy of "read the wiki" should either be applied everywhere (and no comments at all anywhere) or some explanation should be provided here. I'm asking these questions now here instead of 100s of developers asking it later.

Reply:
In my experience, as a  dev I'd go to the included file (which is in the same folder, not abstracted at all) and follow the link to the wiki.

Reply:
I do tend to like one-line comments for existing developers as they trigger a memory of something you already know. That's a bit me I admit.